apps/contracts/contracts/Zengo.sol

```diff
@@ -3,39 +3,54 @@ pragma solidity ^0.8.17;
 
 import "@thirdweb-dev/contracts/extension/PermissionsEnumerable.sol";
 import "@thirdweb-dev/contracts/extension/ContractMetadata.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./Constants.sol";
 
-interface ERC20Token {
-    function transfer(address to, uint256 value) external returns (bool);
-}
-
-contract ZengoDAO is PermissionsEnumerable, ContractMetadata {
+contract ZengoDAO is PermissionsEnumerable, ContractMetadata, Constants {
+    IERC20 public token;
     address public owner;
     uint256 public registrationDuration;
     uint256 public pluralVotingPoints;
     address public votingTokenAddress; // Address of the ERC20 voting token
     address public deployer;
 
+    event ModeratorAdded(address indexed newModerator);
+    event ModeratorRemoved(address indexed removedModerator);
+    event ProposalSubmitted(Proposal newProposal, Evidence newEvidence);
+
     // Modify the Design to facilitate current implementation
     struct Proposal {
+        uint8 votingIterationCount;
         uint256 proposalId;
         string title;
         string proposalDescription;
         string proposalType;
         address proposer;
-        Evidence proposerEvidence;
+        Evidence proposalEvidence;
         Vote[] votingIterations;
+        VerificationState verificationState;
         // requires funding
         bool isEligibleForFunding;
         bool isVerified;
```

Comment:
What represents the `isVerified` bool?  Does it represent the outcome for solved/not valid if it is false and the evidence verification if it is true?

Reply:
``isVerified`` flag indicates that it out of verification state and no further votingIteration would be done

Reply:
> `isVerified` flag indicates that it is out of verification state and no further votingIteration would be required further. It has achieved one of the 3 final states

This boolean could be use for the frontend, to conditionally manage each proposal based on this boolean. Perhaps another name could be appropiate. Something that expresses that you said. For example:
-  `alreadyVerificated` meaning that the VotingIterations already went throught
- `pendingResolution` meaning the true value that any of th 3 final states has achieved yet

just dropping ideas.

Reply:
Yea, I think that's why ``isVerified`` makes sense here